src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1038,7 +1038,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
 
         BuildData previousBuildData = getBuildData(build.getPreviousBuild());   // read only
         BuildData buildData = copyBuildData(build.getPreviousBuild());
-        build.addAction(buildData);
```

Comment:
Right, in copies builddata as soon as it can because previousBuildData may be changed. It copies to build and then allows modify this BuildData.